src/twisted/conch/ssh/keys.py

```diff
@@ -1198,7 +1379,18 @@ def sign(self, data):
             ret = common.NS(int_to_bytes(r, 20) + int_to_bytes(s, 20))
 
         else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+            # Hash size depends on key size
```

Comment:
This should probably be `elif keyType == 'EC'` for clarity, even if that previous `else` block was just dead code :)